### PR TITLE
feat(browser-starifish): make page alerts dismissible

### LINKED ISSUE #64099: Make make sdk version alert dismissable

### PR DESCRIPTION
**Pull request by DominikB2014**
Fixes https://github.com/getsentry/sentry/issues/64099

And more changes post #61881 

1. This adds a `dismissId` property to pageAlert options. When set, the id is used to uniquely identify a dismissible alert so that it can maintain in its dismissed state post reload via localstorage.

2. Implements the alert to the resource SDK alert
image

### CODE REVIEW

File: static/app/utils/performance/contexts/pageAlert.tsx [hunk 1]
```diff
@@ -2,15 +2,23 @@ import React, {createContext, Fragment, useContext, useState} from 'react';
...
 import {Alert} from 'sentry/components/alert';
+import {IconClose} from 'sentry/icons';
+import {useLocalStorageState} from 'sentry/utils/useLocalStorageState';
 
 type PageAlertType = keyof Theme['alert'];
 
 export type PageAlertOptions = {
   message: React.ReactNode | undefined;
   type: PageAlertType;
+  dismissId?: string;
```
Comment (AbhiPrasad) [inline static/app/utils/performance/contexts/pageAlert.tsx:13]: h: Can we make this a enum instead? And make users of this `PageAlert` add to an enum here?

File: static/app/utils/performance/contexts/pageAlert.tsx [hunk 2]
```diff
@@ -68,15 +76,36 @@ export function PageAlertProvider({children}: {children: React.ReactNode}) {
...
+  if (dismissId && dismissedAlerts.includes(dismissId)) {
+    return null;
+  }
+
+  const handleDismiss = () => {
+    if (!dismissId || dismissedAlerts.includes(dismissId)) {
+      return;
+    }
+    const prev = new Set(dismissedAlerts);
+    setDismissedAlerts([...prev, dismissId]);
```
Comment (AbhiPrasad) [inline static/app/utils/performance/contexts/pageAlert.tsx:99]: I think the spread means that we still risk getting duplicates added, and we should only call the setter if we see there was a new `dismissId` added

```suggestion
    const newDismissedAlerts = new Set(dismissedAlerts);
    newDismissedAlerts.add(dismissId);
    
    if (newDismissedAlerts.size !== dismissedAlerts.length) {
      setDismissedAlerts([...newDismissedAlerts]);
    }
```

Comment (DominikB2014) [reply static/app/utils/performance/contexts/pageAlert.tsx:103]: I think this should be already covered with the following code we have already
```
if (!dismissId || dismissedAlerts.includes(dismissId)) {
      return;
    }
```

File: static/app/utils/performance/contexts/pageAlert.tsx (hunk 1 above)
Comment (DominikB2014) [reply static/app/utils/performance/contexts/pageAlert.tsx]: done!

**Review by AbhiPrasad — APPROVED** [review]

**Pull request merged**